DemoApp/BannerShowingConnectionDelegate.swift

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+final class BannerShowingConnectionDelegate {
+    // MARK: - Private Properties
+    
+    private let navigationController: UINavigationController
+    private var connectionEstablishmentTime: Date?
+    
+    // MARK: -
+    
+    init(navigationController: UINavigationController) {
+        self.navigationController = navigationController
+    }
+}
+
+// MARK: - ChatConnectionControllerDelegate
+
+extension BannerShowingConnectionDelegate: ChatConnectionControllerDelegate {
+    public func connectionController(_ controller: ChatConnectionController, didUpdateConnectionStatus status: ConnectionStatus) {
+        switch status {
+        case .connected:
+            connectionEstablishmentTime = Date()
+        case .disconnecting:
+            break
+        case .connecting:
+            if let time = connectionEstablishmentTime {
+                let elapsedTime = time.distance(to: Date())
+                if elapsedTime > 5 {
+                    // TODO: We need to reset the property we make desicion on whether to show the banner or not
```

Comment:
I left this comment because I'm not sure that `.connected` event ever fires second time for example when you recover after a disconnect, but I'm going to double check that